rfcs/0128-selective-auto-merge-on-bot-upgrades.md

```diff
@@ -0,0 +1,53 @@
+---
+feature: Selective auto-merge on bot upgrades
+start-date: 2022-07-07
+author: superherointj
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a `meta.autoMerge` attribute to packages to allow committers to delegate merge rights to package maintainers for bot auto updates.
+
+# Motivation
+[motivation]: #motivation
+
+* Reduce pending PRs for review and merge in nixpkgs.
+* Save reviewers and commiters time.
+* Speed up package upgrades.
+
+# Detailed design
+[design]: #detailed-design
+
+Add a new `meta.autoMerge` package attribute with type `bool` defaulting to `false`. To be documented in Nixpkgs manual.
+
+OfBorg reads `meta.autoMerge` value and sets `13.auto-merge` label to PR.
+
+GitHub Actions will trigger a merge when the following 4 conditions are met:
+1) `@r-ryantm` is PR's author.
+2) Has `13.auto-merge` label.
+3) All CI checks passed.
+4) A package maintainer has approved PR. (Label: `12.approved-by: package-maintainer`)
+
+Due maintainers being able to add labels in GitHub, an extra check for labels might be necessary in GitHub Actions before merge can happen.
```

Comment:
Before we even start to look at the proposed solution, we need to validation if this is even possible.

Reply:
I think you are arguing us into a chicken-and-egg situation. If we can't more or less agree to do the overall thing, I don't think someone should put in the substantial work to do it securely and well. Perhaps the wording could be changed to say only secure implementations (not relying on labels) would be accepted?

Reply:
We can improve wording from:

> Due maintainers being able to add labels in GitHub, an extra check for labels might be necessary in GitHub Actions before merge can happen.

To:

> Due to maintainers being able to add labels in GitHub, an extra check of the 4 conditions is necessary in GitHub Actions before merge can happen. Labels are not to be trusted and information must be checked from a trusted source.

Or something else? Let me know.

Reply:
> Perhaps the wording could be changed to say only secure implementations (not relying on labels) would be accepted?

Yeah, maybe this would be better. I think it is fine to trigger the whole process with labels, however the actual checks needs to be done in a more secure and robust way.

Reply:
Agreed.